README.md

```diff
@@ -1,31 +1,302 @@
-# stolon-pgbouncer
+# stolon-pgbouncer [![CircleCI](https://circleci.com/gh/gocardless/stolon-pgbouncer/tree/master.svg?style=svg)](https://circleci.com/gh/gocardless/stolon-pgbouncer/tree/master)
+'stolon-pgbouncer' extends a highly-available PostgreSQL setup (managed by
+[stolon](https://github.com/sorintlab/stolon)) enabling PgBouncer proxying  and
+zero-downtime failover of the PostgreSQL primary.
 
-## Tests
+See [Playground](#playground) for how to start a Dockerised three node stolon PostgreSQL
+cluster utilising `stolon-pgbouncer`.
 
-Run unit tests:
-    make test
+- stolon-pgbouncer
+    - [Overview](#overview)
+    - [Architecture](#architecture)
+    - [Commands](#commands)
+- [Playground](#playground)
+    - [Architecture](#playground-architecture)
+    - [Installation](#playground-installation)
+    - [Commands](#playground-commands)
 
-Run acceptance tests:
-    make test-acceptance
+- [Development](#development)
+    - [Images](#images)
+    - [Releasing](#releasing)
 
-This will spin up a cluster using docker-compose and run acceptance tests
-against it.  Right now there's a race between the cluster becoming stable and
-the test starting, so it may be a bit flaky.
+### Overview
+GoCardless runs a highly available PostgreSQL cluster using
+[stolon](https://github.com/sorintlab/stolon). It provides us with many of the
+features we need to accomplish this however for our use case and set up we
+required two vital extra features.  Being able to perform zero-downtime failover
+(the stolon-proxy is a layer 4 proxy and can't pause traffic) supporting the
+placing of PgBouncer in front of stolon (as there is no mechanism to update the
+PostgreSQL node that PgBouncer is pointing to when the master moves) are
+important to us. It is from these two requirements that `stolon-pgbouncer` was
+developed.
 
-## Development
+`stolon-pgbouncer` makes use of the stolon clusterview stored in etcd and has
```

Comment:
Going to try this without explaining where our services run, in case it makes it read lighter